### ISSUE TITLE
Requests to this API openidconnect.googleapis.com method google.identity.oauth2.openidconnect.v1.OpenIdConnectService.GetLegacyUserInfo are blocked.

### ISSUE THREAD

**Issue by nightsurge** [claudiob/yt#388]
I have made no changes to my implementation in months, yet all of the sudden I am getting these errors a lot. I cannot authenticate with my app any longer, as a result...

In my code, this error fires when accessing "Account.avatar_url", after having passed in the authorization_code and redirect_uri, and gotten the access_token and refresh_token.

Are there changes happening via Google APIs or something recently? Can anyone help me figure out whether this is a Gem/code issue or if something has changed that I need to update in my Google Console?

```ruby
Yt::Errors::Forbidden: A request to YouTube API was considered forbidden by the server:
{"error"=>{"code"=>403, "message"=>"Requests to this API openidconnect.googleapis.com method google.identity.oauth2.openidconnect.v1.OpenIdConnectService.GetLegacyUserInfo are blocked.", "status"=>"PERMISSION_DENIED", "details"=>[{"@type"=>"type.googleapis.com/google.rpc.Help", "links"=>[{"description"=>"Google developer console API key", "url"=>"https://console.developers.google.com/project/MYPROJECTNUMBER/apiui/credential"}]}]}}

You can retry the same request manually by running:
curl -X GET -H "content-length: 0" -H "user-agent: Yt::Request (gzip)" -H "authorization: Bearer MYBEARERTOKENLOL" -H "host: www.googleapis.com" "https://www.googleapis.com/oauth2/v2/userinfo?key=MYKEYLOL"
```

**Comment by nightsurge** [comment]
Is it possible that with the retiring of certain Google+ APIs and whatnot, that this gem is still trying to pull in data from the wrong place? Did I miss a big change in how to setup and use this gem that happened in the last few years, that my app will need updated to use?

**Comment by dgb** [comment]
Hmm, I'm not sure you can access that endpoint with an API key. I'd think you'd need OAuth credentials to do so. What sort of thing are you doing in Yt that's causing that error to occur?

**Comment by nightsurge** [comment]
@dgb in my code specifically, the error reports occur when I have retrieved the Account using the authorization code, and then try to access the avatar_url.

Based on the snippet I posted above, though, isn't it including the OAuth into the request as well, in the Bearer token? (I replaced it for security reasons)

I really can't think of any reason why this is failing unless Google made changes to their APIs at some point or something. I have a verified/published app and made no changes to my code in the last several months, yet this issue just kind of started out of no where.

**Comment by kangkyu** [comment]
I'm curious what is `scopes` for your account @nightsurge

**Comment by nightsurge** [comment]
@kangkyu the 3 default scopes of email, profile, openid, and then "youtube", "youtube.readonly", and "youtube.upload" for the YT gem functionality I have been using for years now. They all have green check marks and are verified.

**Comment by nightsurge** [comment]
@kangkyu @dgb should I be using "userinfo.email" and "userinfo.profile" for those, or just "email" and "profile"?

Example:
`https://www.googleapis.com/auth/userinfo.email`
vs
`https://www.googleapis.com/auth/email`

**Comment by kangkyu** [comment]
@nightsurge the first one https://developers.google.com/identity/protocols/oauth2/scopes#oauth2
by the way do you have all those scopes in your "Scopes for Google APIs" in "OAuth consent screen" page?

**Comment by nightsurge** [comment]
@kangkyu Yes, the email, profile, openid ones are there by default, and the 3 youtube ones are there with Green Checkmarks. I already went through the verification a year ago or so.

**Comment by nightsurge** [comment]
<img width="471" alt="Screen Shot 2020-06-09 at 8 18 38 AM" src="https://user-images.githubusercontent.com/3065882/84146564-e1638c00-aa29-11ea-89d3-469c0a4b6278.png">

Thoughts? Here is my authorization URI:
```text
'https://accounts.google.com/o/oauth2/auth?access_type=offline&approval_prompt=force&client_id=MYCLIENTID&redirect_uri=https%3A%2F%2Fgamerdvr.com%2Fyt%2Fauth%2Fsuccess&response_type=code&scope=https%3A%2F%2Fwww.googleapis.com%2Fauth%2Fyoutube+https%3A%2F%2Fwww.googleapis.com%2Fauth%2Fyoutube.readonly+https%3A%2F%2Fwww.googleapis.com%2Fauth%2Fyoutube.upload+https%3A%2F%2Fwww.googleapis.com%2Fauth%2Fuserinfo.profile+https%3A%2F%2Fwww.googleapis.com%2Fauth%2Fuserinfo.email'
```

**Comment by dgb** [comment]
@nightsurge sorry, I missed the OAuth bearer token in your initial issue posting. I'd initially thought you were using just the API key to access that endpoint. Then I thought there might be an issue using both the API key and the bearer token, but I tried with one of our apps and it seems to work fine.

Based on the error message I'd think that the authentication tokens were created before the permission was granted to the underlying profile resource, or permission was revoked. I (vaguely) remember Google doing some work on how profiles were represented, so there could be an issue there, but I don't know specifically.

In any case, I'm pretty sure this is an issue on the Google account side of things. I'd try either using a different account or revoking permissions from the Google account side of things, re-authing, and seeing if the issue persists.

**Comment by nightsurge** [comment]
@dgb I just tried it again with a different account... getting the same errors. I haven't tried it with a different API/project yet.. I just can't see for the life of me why this is happening. Another oddity I see now in my Google APIs auth screen settings is a 98/100 User Cap... but that has not changed to reach 100 in a few days now.

I'm definitely inclined to believe this is somehow related to Profiles/Auths, but I can't figure out what that may be, and for it to just start happening out of the blue like that...

**Comment by nightsurge** [comment]
One other fun thing I've noticed recently... My quotas and error logs show the Youtube Data API twice?
<img width="1014" alt="Screen Shot 2020-06-09 at 2 52 49 PM" src="https://user-images.githubusercontent.com/3065882/84188065-f871a080-aa60-11ea-9877-84c0e3fa9d39.png">
<img width="554" alt="Screen Shot 2020-06-09 at 2 54 57 PM" src="https://user-images.githubusercontent.com/3065882/84188200-34a50100-aa61-11ea-8d32-a64389e55128.png">

**Comment by nightsurge** [comment]
So just confirming, but are you both able to hit the "account.avatar_url" without any failures? @dgb @kangkyu 

I tested with a new google_oauth gem for getting the authentication piece, and passing in the access token I got from that. The calls work when getting videos, etc. but when I specifically try to hit "avatar_url" I get the error from the original message still. Seems like maybe it is an outdated endpoint? The new authentication with omniauth-google_oauth2 gem returns the image as part of that initial auth so I don't need the avatar_url anymore.
